### DISCUSSION TITLE
Create custom UI

### DISCUSSION THREAD

**Discussion by levyitay** [mitmproxy/mitmproxy#6414] [Q&A]
Hi All, 
I'm trying to implement custom webserver to work with Mitmproxy, but my WS keeps returning 403. 
Is there a proper way to setup the proxy to accept different WS requests which are not originated from it's own web server?

**Comment by mhils** [comment]
You probably need to enable CORS requests in our Tornado web server. AFAIK that's only possible by modifying mitmproxy sources or by giant monkeypatch hackery. Alternatively, you could host your front end on the same origin.

**Reply by Prinzhorn** [reply]
> You probably need to enable CORS requests in our Tornado web server.

CORS does not apply to WS :wink: , the host check needs to be done separately via `check_origin` (I couldn't find any reference in our source though?). Also https://github.com/mitmproxy/mitmproxy/issues/3393

**Reply by mhils** [reply]
Yeah, `check_origin` is a Tornado thing, see [here](https://www.tornadoweb.org/en/stable/websocket.html#tornado.websocket.WebSocketHandler). That's what I meant by "enable CORS requests in our Tornado web server". You are of course technically correct (the best kind of correct) that it's not CORS. 😉

**Reply by levyitay** [reply]
Found it!
What would be the correct way to build linux / Mac executable and test it?

**Reply by levyitay** [reply]
@mhils @Prinzhorn ?

**Comment by levyitay** [comment]
Here is the PR:
https://github.com/mitmproxy/mitmproxy/pull/6443
